AppInfoView: redesign links

danirabbit opened the pull request:
Fixes #1134

Show more links, add some color. Group links related to contributing/giving vs requesting/contacting/receiving

![Screenshot from 2024-06-21 11 46 44](https://github.com/elementary/appcenter/assets/7277719/122a8a5d-6fec-4cb1-90df-685d53ddd0be)
![Screenshot from 2024-06-21 11 47 37](https://github.com/elementary/appcenter/assets/7277719/fa9dd68e-86b3-412e-9eb0-814571c35ece)

ryonakano reviewed (commented):
Really cute and nice improvements! Leaving some comments.

ryonakano commented on src/Widgets/LinkListBox.vala:
```diff
@@ -0,0 +1,286 @@
+            flowbox.append (contact_listbox);
+        }
+
+        // Don't let invisible container get focus
+        var flowbox_child = flowbox.get_first_child ();
+        while (flowbox_child != null) {
+            flowbox_child.focusable = false;
+            flowbox_child = flowbox_child.get_next_sibling ();
+        }
```
I'm not sure what this actually does. The comment says that this prevents invisible flowboxes from being focused but the code doesn't check for the visibility of flowboxes.

danirabbit replied:
Updated! Is that better?

ryonakano replied:
Thank you, now I understand what we're doing here.

ryonakano commented:
Also, there are extra margin right to the links if contribute_listbox is not appened.

![Screenshot from 2024-06-23 11-28-00](https://github.com/elementary/appcenter/assets/26003928/92ae8337-5933-48e6-b189-f82e2427a008)

leolost2605 commented on src/Widgets/LinkListBox.vala:
```diff
@@ -0,0 +1,286 @@
+            };
+
+            stripe_dialog.download_requested.connect (() => {
+                if (stripe_dialog.amount != 0) {
```
This will cause a mem leak :(
(Using a local variable inside a lambda that's connected to a signal)

danirabbit replied:
Moved this bit to the stripe dialog itself which is more DRY anyways

danirabbit commented on src/Views/AppInfoView.vala:
```diff
@@ -1185,80 +1095,6 @@ public class AppCenter.Views.AppInfoView : Adw.NavigationPage {
-    class UrlButton : Gtk.Box {
-        public UrlButton (string label, string? uri, string icon_name) {
-            add_css_class (Granite.STYLE_CLASS_DIM_LABEL);
-            tooltip_text = uri;
```
I decided to add the description label directly to the box as GNOME Software is doing here so that it's also accessible for touch

danirabbit commented:
Updated! I added description labels directly to the box here and also made sure to update the icon and a11y for stripe buttons. You can see here how it differs between a link and a payment that's handled by stripe:

![Screenshot from 2024-06-24 18 17 39](https://github.com/elementary/appcenter/assets/7277719/9b322288-d153-4e43-a493-71bd90d9da0a)
![Screenshot from 2024-06-24 18 20 19](https://github.com/elementary/appcenter/assets/7277719/14d60a14-2863-4949-8345-f068b3acabe9)
![Screenshot from 2024-06-24 18 22 42](https://github.com/elementary/appcenter/assets/7277719/a6097e6d-b695-4982-bcf1-09613cbc87ab)

leolost2605 reviewed (approved):
Looks good from my side :)

ryonakano reviewed (approved):
Maybe we can create the LinkRow class in Granite for consistent design, because the same design is used in https://github.com/elementary/switchboard-plug-about/pull/325 too, but we can discuss that later.
